File: Data Structures & Algorithms/max-water-container/submission-0.js
class Solution {
    /*** @param {number[]} 
     * heights* 
     * @return {number}*/
    maxArea(heights) {
        let right = heights.length-1;
        let left = 0;
        let max = 0;

        while(left<=right) {
            const height = Math.min(heights[left], heights[right]);
            const width = right - left;
            const currentArea = height * width;

            max = Math.max(max, currentArea);

            if (heights[left] < heights[right]) {
                left++;
            } else {
                right--;
            }
        }

        return max;

    }

}

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        k = 0
        minK =  max(piles)
        left = 1
        right = max(piles)

        while(left<=right):
            currentK = left + ((right-left) // 2)
            currentH = 0
            for pile in piles:
                currentH += math.ceil(pile/currentK)
            
            if(currentH > h):
                left = currentK + 1
            elif (currentH <= h):
                right = currentK - 1 
                minK = min(minK, currentK)


        return minK